lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4908,14 +5016,19 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
         auto activeValueAdj = getAdjointValue(origBB, activeValue);
         auto concreteActiveValueAdj =
             materializeAdjointDirect(activeValueAdj, loc);
-        trampolineArguments.push_back(concreteActiveValueAdj);
+
+        if (!adjointTrampolineBlockMap.count(concreteActiveValueAdj)) {
```

Comment:
This logic is hacky and is not straightforward to describe. Fortunately it will be defined away with the upcoming [ad-all-indirect](https://github.com/rxwei/swift/tree/ad-all-indirect) that @dan-zheng is working on.